test/test_desparsified_lasso.py

```diff
@@ -102,38 +115,125 @@ def test_desparsified_group_lasso():
-    np.testing.assert_raises(
-        ValueError, desparsified_lasso, X=X, y=Y, multioutput=True, covariance=bad_cov
+    # np.testing.assert_raises(
+    # ValueError, desparsified_lasso, X=X, y=y, multioutput=True, covariance=bad_cov
+    # )
+    desparsified_lasso = DesparsifiedLasso(
+        model_y=multitasklassoCV, covariance=bad_cov
+    ).fit(X, y)
+    with pytest.raises(ValueError):
+        desparsified_lasso.importance()
+
+    with pytest.raises(AssertionError, match="Unknown test 'r2'"):
+        DesparsifiedLasso(model_y=multitasklassoCV, covariance=bad_cov, test="r2").fit(
+            X, y
+        )
+
+
+def test_exception():
+    n_samples = 50
+    n_features = 100
+    n_target = 10
+    support_size = 2
+    signal_noise_ratio = 50
+    rho_serial = 0.9
+    corr = toeplitz(np.geomspace(1, rho_serial ** (n_target - 1), n_target))
+    multitasklassoCV = MultiTaskLassoCV(
+        eps=1e-2,
+        fit_intercept=False,
+        cv=KFold(n_splits=5, shuffle=True, random_state=0),
+        tol=1e-4,
+        max_iter=5000,
+        random_state=1,
+        n_jobs=1,
     )
 
+    X, y, beta, noise = multivariate_simulation(
+        n_samples=n_samples,
+        n_features=n_features,
+        n_targets=n_target,
+        support_size=support_size,
+        rho_serial=rho_serial,
+        signal_noise_ratio=signal_noise_ratio,
+        seed=10,
+    )
+
+    with pytest.raises(
+        AssertionError, match="lasso needs to be a Lasso or a MultiTaskLasso"
+    ):
+        DesparsifiedLasso(model_x=RandomForestClassifier())
+    with pytest.raises(
+        AssertionError, match="lasso_cv needs to be a LassoCV or a MultiTaskLassoCV"
+    ):
+        DesparsifiedLasso(model_y=RandomForestClassifier())
+    with pytest.raises(AssertionError, match="Unknown test 'r2'"):
+        DesparsifiedLasso(test="r2")
+    desparsified_lasso = DesparsifiedLasso(model_y=multitasklassoCV)
+    with pytest.raises(
+        ValueError,
+        match="The Desparsified Lasso requires to be fit before any analysis",
+    ):
+        desparsified_lasso.importance()
+    desparsified_lasso.sigma_hat_ = []
+    with pytest.raises(
+        ValueError,
+        match="The Desparsified Lasso requires to be fit before any analysis",
+    ):
+        desparsified_lasso.importance()
+
+    desparsified_lasso = DesparsifiedLasso(model_y=multitasklassoCV).fit(X, y)
     with pytest.raises(ValueError, match="Unknown test 'r2'"):
-        desparsified_group_lasso_pvalue(beta_hat, theta_hat, precision_diag, test="r2")
+        desparsified_lasso.test = "r2"
+        desparsified_lasso.importance()
+
+    desparsified_lasso = DesparsifiedLasso(model_y=multitasklassoCV).fit(X, y)
+    with pytest.warns(Warning, match="X won't be used."):
+        desparsified_lasso.importance(X=X)
+    with pytest.warns(Warning, match="y won't be used."):
+        desparsified_lasso.importance(y=y)
+
+
+def test_function_not_center():
+    "Test function and not centered"
```

Comment:
WDYM?

Reply:
Test the usage of the option center=False.